guides/releasing/Release Workflow.md

```diff
@@ -0,0 +1,328 @@
+
+**How:**
+
+- On a new branch, check all `expo-template-*` packages under `templates` directory and bump dependencies versions wherever possible. Use versions stored in `packages/expo/bundledNativeModules.json` for vendored libs like `react-native-gesture-handler`.
+- Run `et publish-templates` and answer to questions it asks. Prerelease versions should be tagged as `next` and not `latest`.
+- Create a pull request from your branch to `master`. Make sure a reviewer will cherry-pick that commit to the release branch as well.
+
+# Stage 4 - Expo client
+
+## 4.1. Cutting off release branch
+
+| Requirements                                                                                 |
+| -------------------------------------------------------------------------------------------- |
+| [3.3. Publishing prerelease project templates](#3-3-publishing-prerelease-project-templates) |
+
+**Why:** Sometimes we need to apply changes to some older SDKs (roll out them to the stores or Turtle builders, publish patches to NPM registry). That's why we need to keep the current repository state on a separate, release branch so changes made to `master` in the meantime won't interfere.
+
+**How:** Once both iOS and Android versioning pull requests are merged, you can cut off `sdk-XX` branch from `master` and push it to the remote repo.
+
+## 4.2. Releasing beta version
+
+| Requirements                                                       |
+| ------------------------------------------------------------------ |
+| [4.1. Cutting off release branch](#4-1-cutting-off-release-branch) |
+
+**Why:**
+
+**How:**
+
+- **iOS**:
+  - Bump Expo client versions (CFBundleVersion, CFBundleShortVersionString) in `ios/Exponent/Supporting/Info.plist`.
+  - As of latest time of writing, Xcode's "automatic provisioning" isn't working well with fastlane, so you might want to switch to manual provisioning in your working copy of Exponent.xcodeproj.
+  - Apple seems to invalidate our provisioning profiles whenever somebody new gets added or removed from the team. If you are using manual provisioning, it is sometimes helpful to log in, clear invalid profiles, and create/download a new one. If you do this, please name the profile something very specific, like 'Expo Client App Store August 20 2018'!
+  - Make sure build's metadata are up to date (see files under `fastlane/metadata/en-US`).
+  - Run `fastlane ios release` from the project root folder and follow the prompt. This step can take 30+ minutes, as fastlane will update (or create) the App Store Connect record, generate a signed archive, and upload it.
+  - Wait for Apple to finish processing your new build. This step can take another 30+ minutes (but sometimes just a few).
+  - Publish that build to TestFlight and send invitations to other testers. You should also do some smoke tests, for example against `native-component-list` published under `applereview` account.
+
```

Comment:
Android:
- wait for iOS to be approved!
- Bump the `versionCode` and `versionName` in android/app/build.gradle. Commit this to master and cherry-pick to the release branch. You might need to check the previous release branch to make sure the new `versionCode` is greater than the previous patch version, in case that commit never made it to master.
- Add a changelog for the new version in `fastlane/android/metadata/en-US/changelogs`. Commit to master and cherry-pick to the release branch.
- Find the `client_android` CI job on the release branch. When it completes, download the APK from Artifacts and do a smoke test -- install it on a fresh Android device, turn on airplane mode, and make sure Home loads.
- On the release branch, approve the `client_android_apk_release` build job.
- When ready to release to production, approve the `client_android_release_google_play` build job. Note that *this will release the new client in the Play Store* so only do this when everything else is ready!